graphql_compiler/tests/test_post_processing.py

```diff
@@ -0,0 +1,221 @@
+# Copyright 2019-present Kensho Technologies, LLC.
+import datetime
+from unittest import TestCase
+
+from ..post_processing.sql_post_processing import post_process_mssql_folds
+from .test_helpers import get_sqlalchemy_schema_info
+
+
+class MssqlXmlPathTests(TestCase):
+    def setUp(self) -> None:
+        self.mssql_schema_info = get_sqlalchemy_schema_info(dialect="mssql")
+
+    def test_convert_empty_string(self):
+        """Test empty list is correctly decoded."""
+        query_output = [{"child_names": "",}]
+        graphql_query = """{
+                    Animal {
+                        in_Animal_ParentOf @fold{
+                            name @output(out_name: "child_names")
+                        }
+                    }
+                }"""
+
+        expected_result = [{"child_names": []}]
+
+        post_process_mssql_folds(self.mssql_schema_info, graphql_query, query_output)
+        self.assertEqual(query_output, expected_result)
+
+    def test_convert_basic(self):
+        """Test basic XML path encoding (only pipe separations) is correctly decoded."""
+        query_output = [{"child_names": "Animal 1|Animal 2|Animal 3|",}]
```

Comment:
Can you put two consecutive empty string elements in the middle? Just protecting against different string splitting methods (I can imagine a sensible one that merges multiple occurances of the delimiter).